### PR TITLE
RMMIS-5106: Add dependency type of 'enablement' that will disable/enable field, like visibility type.

### PR DESCRIPTION
**Pull request by traviscooperscdhhs**

### CODE REVIEW

File: test/unit/dependency-mixin-spec.js [hunk 1]
```diff
@@ -16,48 +16,96 @@ var config = {
...
   });
 
   it('can update a field\'s visibility when a dependent field changes its value', function(done){
     expect(field.state.visible).toEqual(false);
     Flux.doAction(constants.actions.FIELD_VALUE_CHANGE, {name: 'dep-mixin-test', value: 'bar'})
-      .then(function(){
-        expect(field.state.visible).toEqual(true);
-        done();
-      });
+        .then(function(){
```
Comment (mallenscdhhs) [inline test/unit/dependency-mixin-spec.js]: extra whitespace

File: test/unit/dependency-mixin-spec.js [hunk 2]
```diff
@@ -16,48 +16,96 @@ var config = {
...
 
   // simulates listening for value change on a checkbox group, which has an Array as its value
   it('can determine if a field\'s state should change when a list of values are sent', function(done){
     expect(field.state.visible).toEqual(false);
     Flux.doAction(constants.actions.FIELD_VALUE_CHANGE, {name: 'dep-mixin-test', value: ['boop', 'bar']})
-      .then(function(){
-        expect(field.state.visible).toEqual(true);
-        done();
-      });
+        .then(function(){
```
Comment (mallenscdhhs) [inline test/unit/dependency-mixin-spec.js]: extra whitespace

File: test/unit/dependency-mixin-spec.js [hunk 3]
```diff
@@ -16,48 +16,96 @@ var config = {
...
-  
+
   it('determine if a field should be visible based on initial value of dependent field', function(done){
     expect(field.state.visible).toEqual(false);
     Flux.doAction(constants.actions.FIELD_VALUE, {name: 'dep-mixin-test', value: ['boop', 'bar']})
-      .then(function(){
-        expect(field.state.visible).toEqual(true);
-        done();
-      });
+        .then(function(){
```
Comment (mallenscdhhs) [inline test/unit/dependency-mixin-spec.js]: extra whitespace

File: src/DependencyMixin.js [hunk 4]
```diff
@@ -39,18 +43,28 @@ module.exports = {
   componentDidMount: function(){
     if(this.hasDependency()){
       var visible = this.state.visible;
+      var disabled = this.state.disabled;
       var depName = this.props.dependencyName;
+      var depType = this.props.dependencyType || '';
```
Comment (mallenscdhhs) [inline src/DependencyMixin.js]: I don't like having `dependencyType` being a passed in prop. This should be derived from the `initialState` value instead.

File: test/unit/dependency-mixin-spec.js (hunk 1 above)
Comment (traviscooperscdhhs) [inline test/unit/dependency-mixin-spec.js]: what in the world? where did that whitespace come from.  

I'll get that taken care of.

File: src/DependencyMixin.js (hunk 4 above)
Comment (traviscooperscdhhs) [inline src/DependencyMixin.js]: Cool.  I'll have it look for 'enabled' or 'disabled' in initialState.  Makes it more backwards compatible as well.

**Comment by kjschulz** [comment]
:+1:

File: src/DependencyMixin.js [hunk 5]
```diff
@@ -39,18 +43,28 @@ module.exports = {
...
-      Dispatcher.register( this.props.id + '-DEP-CHANGE' , function(action,data){
+      Dispatcher.register( this.props.id + '-DEP-CHANGE' , (action,data)=>{
         if( _.includes([constants.actions.FIELD_VALUE_CHANGE,constants.actions.FIELD_VALUE],action) &&
           data.name === depName){
           var value = _.isArray(data.value)? data.value : [data.value];
-          var visibility = (utils.containsOneOf(depValues, value))? !visible : visible;
-          this.setState({'visible': visibility});
-          Flux.doAction(constants.actions.COMPONENT_VISIBILITY,this.props,visibility);
+          var changeState = utils.containsOneOf(depValues, value);
+          if(depType === 'enablement'){
```
Comment (mallenscdhhs) [inline src/DependencyMixin.js]: Im not sure that we need to check for `depType` at all. Couldn't you just do:
```
let disabledState = changeState? !disabled : disabled;
let visibilityState = changeState? !visible : visible;
this.setState({visible: visibilityState, disabled: disabledState});
```

Also, what is the need for `COMPONENT_VISIBILITY` action?

Comment (traviscooperscdhhs) [inline src/DependencyMixin.js]: COMPONENT_VISIBILITY is used to keep track of component visibility by other fields.

**Pull request merged**